### ISSUE TITLE
Provide convenience event to enable/disable picking

### ISSUE THREAD

**Issue by dimvoly** [aevyrie/bevy_mod_picking#169]
Hi there, really good plug in, thanks for developing it.

In my situation. at times, I need for nothing to be pickable, and then at other times I need the pickability to come back. I managed to do this by adding/removing the `PickableBundle` components. One issue I came across was that if pickability was removed, any meshes that were highlighted were still highlighted. I did manage to make it work by looking through the source code and changing the materials back to that saved on the `Highlighting` component. See code below that listens for events.

    pub enum TogglePicking {
	    EnablePicking,
	    DisablePicking,
	}

	fn toggle_picking_events(
	    mut events: EventReader<TogglePicking>,

	    all_my_pickables_pickable_query: Query<
		Entity,
		(
		    With<MyPickableEntity>,
		    With<PickableMesh>,
		    With<Interaction>,
		    With<FocusPolicy>,
		    With<Highlight>,
		    With<Selection>,
		    With<Hover>,
		),
	    >,

	    mut all_with_highlighting_query: Query<(
		Entity,
		&Highlighting<StandardMaterial>,
		&mut Handle<StandardMaterial>,
	    )>,

	    all_my_pickables_unpickable_query: Query<
		Entity,
		(
		    With<MyPickableEntity>,
		    Without<PickableMesh>,
		    Without<Interaction>,
		    Without<FocusPolicy>,
		    Without<Highlight>,
		    Without<Selection>,
		    Without<Hover>,
		),
	    >,
	    mut commands: Commands,
	) {
	    for event in events.iter() {
		match event {
		    TogglePicking::EnablePicking => {
		        // add pickable bundle to all the MyPickable entities
		        for entity in all_my_pickables_unpickable_query.iter() {
		            commands
		                .entity(entity)
		                .insert_bundle(PickableBundle::default());
		        }
		    }
		    TogglePicking::DisablePicking => {
		        for (_, highlighting, mut entity_material) in all_with_highlighting_query.iter_mut()
		        {
		            *entity_material = highlighting.initial.clone();
		        }

		        for entity in all_my_pickables_pickable_query.iter() {
		            // then remove all the Components that make them pickable
		            commands.entity(entity).remove_bundle::<PickableBundle>();
		        }
		    }
		}
	    }
	}

However, if this was part of `bevy_mod_picking`, I think it could be much cleaner than what I've come up with. Up to you.

**Comment by dimvoly** [comment]
To some extent this resource controls some of the above, sorry only just figured this out.

    #[derive(Clone, Debug, Resource)]
    pub struct PickingPluginsState {
        pub enable_picking: bool,
        pub enable_highlighting: bool,
        pub enable_interacting: bool,
    }

I'll be playing with the source code by cloning it, so that I can add some features that are quite specific to my project. If my  modifications are up to scratch, I'll consider doing a pull request. Unless someone gets there first of course.

**Comment by dimvoly** [comment]
I've looked into it further and seems the approach in my first comment could be a lot better. In my use case I need these features:

1. Deselect and unhighlight everything (via an Event)
2. Have nothing be pickable but keep the highlighting and keep firing the hover events
3. Have nothing be pickable, don't highlight anything, but keep firing the hover events

I've forked the code locally for now and have added this to the resource to let the user of the library toggle item 1:





I'm still testing it of course, so if it gets to a point where I'm more confident with it, I'll do a PR.
